text/0032-distinct-attribute.md

```diff
@@ -0,0 +1,238 @@
+
+It is difficult to find the information in the official documentation. A lot of people are asking about the distinct feature on StackOverflow or ElasticSearch forums.
+
+By specifying the search with terms or composite aggregations it is possible to have buckets fed by documents that have the same values on a specified field.
+
+E.g Terms aggregation
+```json
+{
+    "aggs": {
+        "distinct_colors": {
+            "terms": {
+                "field": "color",
+                "size": 1000
+            }
+        }
+    }
+}
+```
+The size parameter can be set to define how many term buckets should be returned out of the overall terms list. Term aggregations by default return 10 buckets only.
+
+Composite aggregation allows to paginate over buckets containing a lot of values.
+
+> Cardinality aggregation can calculates the count of distinct values for a field.
+
+It is also possible to use the keyword `DISTINCT` from the SQL access feature from X-Pack. However the functionality only allows to return tabular data.
+
+### IV.Explanation
+
+Let's say that we have 2 documents with the same `product_id`. Each document exists to materialize the color variation.
+
+```json
+{
+    "hits": [
+        {
+            "colors": "red",
+            "id": 1,
+            "label": "t-shirt",
+            "product_id": 1
+        },
+        {
+            "colors": "black",
+            "id": 2,
+            "label": "t-shirt",
+            "product_id": 1
+        }
+    ],
+    "offset": 0,
+    "limit": 20,
+    "nbHits": 2,
+    "exhaustiveNbHits": false,
+    "processingTimeMs": 1,
+    "query": "t-shirt"
+}
+```
+
+Without setting `product_id` as a distinct attribute, a search with `t-shirt` as a query will return the two documents.
+
+It can be useful to display one product per color variation as a search result for example. But as your number of products variations grows over time, you might want to display only one result as a top search result, mostly for UI concerns.
+
+It's in this case that the distinct attribute finds all its interest.
+
+Setting `product_id` as a distinct attribute will discard all others documents having the same value for `product_id` from the search result.
+
+#### Distinct Attribute in v0.20
+
+> MeiliSearch accepts only one value which is the document attribute that needs to be de-duplicated from the other document sharing the same attribute value.
+
+It is possible to configure the distinct attribute using two endpoints. [Update All Settings](https://docs.meilisearch.com/reference/api/settings.html#update-settings) and the [distinct attribute](https://docs.meilisearch.com/reference/api/distinct_attribute.html#update-distinct-attribute) endpoint.
+
+Given this setting :
+```json
+{
+    "distinctAttribute": "product_id"
+}
+```
+
+MeiliSearch will return de-duplicated hits by `product_id`.
```

Comment:
Not sure I understand `de-duplicated`

Reply:
If we have multiple documents containing the same value for `product_id`. Only one document will be returned if `product_id` is set as a distinct attribute.